# Description of `Students_Take_Tests` table

In [1]:
import pandas as pd
import seaborn as sns

In [2]:
df = pd.read_csv('../tables_creation/final_tables/students_take_tests.csv')
df.rename(columns = {'EDRPOU':'EDRPOU_test_centers'}, inplace=True)
df.head()

outid  year  score100  score12  score  \
0  00000AC7-CDDE-4C77-B979-8B0351AF1305  2017     161.0      9.0    NaN   
1  00000dce-36de-4d58-9dc2-7ffc824f597a  2021       NaN      NaN    NaN   
2  00001a8d-fff5-4c7c-bea2-b0157f7c5655  2021     128.0      6.0   31.0   
3  0000268f-9fdd-49b2-9ee2-422778c9c4f1  2016     160.0      8.0    NaN   
4  0000324e-f525-49c4-a963-8df0cc02d6d5  2018     166.0      9.0   69.0   

        test_status test_subject test_type   KOATUU_2020  EDRPOU_test_centers  
0          Accepted      ukrtest       EIE  5.110137e+09           20995060.0  
1            Absent      ukrtest       EIE  1.810700e+09           22061344.0  
2          Accepted      ukrtest       EIE  7.110136e+09           25922746.0  
3  Received results      ukrtest       EIE  6.310137e+09           24486622.0  
4          Accepted      ukrtest       EIE  4.611200e+09           34387362.0

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10597976 entries, 0 to 10597975
Data columns (total 10 columns):
 #   Column               Dtype  
---  ------               -----  
 0   outid                object 
 1   year                 int64  
 2   score100             float64
 3   score12              float64
 4   score                float64
 5   test_status          object 
 6   test_subject         object 
 7   test_type            object 
 8   KOATUU_2020          float64
 9   EDRPOU_test_centers  float64
dtypes: float64(5), int64(1), object(4)
memory usage: 808.6+ MB


In [4]:
# print unique values for each column
for col in df.columns:
    print('Unique length in ', col+':\t', len(df[col].unique()))

Unique length in  outid:	 2490052
Unique length in  year:	 8
Unique length in  score100:	 117
Unique length in  score12:	 13
Unique length in  score:	 106
Unique length in  test_status:	 8
Unique length in  test_subject:	 14
Unique length in  test_type:	 2
Unique length in  KOATUU_2020:	 634
Unique length in  EDRPOU_test_centers:	 2111


In [5]:
# min, mean and max values for scores columns
for col in df.columns:
    if 'score' in col:
        print(col, '\t', df[col].min(), '\t', round(df[col].mean(),2), '\t', df[col].max())

score100 	 0.0 	 127.51 	 200.0
score12 	 1.0 	 6.67 	 12.0
score 	 0.0 	 32.49 	 104.0


Features:
- outid (Student ID): 2490052 unique  students identifiers
- year (Year): year when student pass test. Range of values:
    - 2018-2023 (integer)
- score100 (Test score): 117 unique score of student in 100-point scale. Range of values:
    - N/A: when student didn't pass this test
    - 100-200 (integer)
- score12 (Test Score in school 12-point scale): 13 unique score of student in 12-point scale. Range of values:
    - N/A: when student didn't pass this test
    - 1-12 (integer)
- score (Raw test score). 106 unique scores values. Different scale for each subject. Range of values:
    - N/A: when student didn't pass this test
    - 0-104 (integer)
- test_status:
    - Accepted: when student passed the test
    - Absent: when student didn't appear on the test
    - Received results: when student passed the test and received results
    - Did not pass the threshold: when student passed the test, but didn't pass the threshold
    - Failed: when student passed the test, but failed
    - Canceled: when student passed the test, but canceled the results
    - Not selected 100-200: when student passed the test, but didn't get 100-200 points
    - Not registered for the main session: when student didn't appear on the test
- test_type (type of test): Range of values:
    - EIE: test which was held in 2016-2021 years
    - NMT: test which was held in 2022-2023 years
- test_subject (name of test subject): Range of values:
    - umltest: Ukrainian language and literature test
    - ukrsubtest: Ukrainian language test only for school graduates
    - ukrtest: Ukrainian language test 
    - mathsttest: Standard math test 
    - mathtest: Math test
    - histtest: History test
    - biotest: Biology test
    - geotest: Geography test
    - chemtest: Chemistry test
    - phystest: Physics test
    - engtest: English language test
    - fratest: French language test
    - deutest: German language test
    - spatest: Spanish language test
    - rustest: russian language test
- KOATUU_2020 (KOATUU code): 634 unique codes(a unique digital code for each administrative-territorial entity of Ukraine, used from January 1998 till January 2021): territory, city area, other countries city for test centers. The first two digits signify the region, the next two represent the district, and the following two denote the city or municipality.
- EDRPOU(unique code of the legal entity or physical person): 2111 unique codes of legal entities, who are owners of test centers(unique test centers code).